selene-lib/src/lints/restricted_imports.rs

```diff
@@ -0,0 +1,95 @@
+use super::*;
+use crate::ast_util::name_paths::name_path;
+use std::collections::HashMap;
+use std::convert::Infallible;
+
+use full_moon::{
+    ast::{self, Ast},
+    visitors::Visitor,
+};
+use serde::Deserialize;
+
+#[derive(Clone, Default, Deserialize)]
+#[serde(default)]
+pub struct RestrictedImportsConfig {
+    pub restricted_paths: HashMap<String, String>,
+}
+
+pub struct RestrictedImportsLint {
+    config: RestrictedImportsConfig,
+}
+
+impl Lint for RestrictedImportsLint {
+    type Config = RestrictedImportsConfig;
+    type Error = Infallible;
+
+    const SEVERITY: Severity = Severity::Error;
+    const LINT_TYPE: LintType = LintType::Correctness;
+
+    fn new(config: Self::Config) -> Result<Self, Self::Error> {
+        Ok(RestrictedImportsLint { config })
+    }
+
+    fn pass(&self, ast: &Ast, _: &Context, _: &AstContext) -> Vec<Diagnostic> {
+        let mut visitor = RestrictedImportsVisitor {
+            restricted_paths: &self.config.restricted_paths,
+            violations: Vec::new(),
+        };
+
+        visitor.visit_ast(ast);
+
+        visitor.violations
+    }
+}
+
+struct RestrictedImportsVisitor<'a> {
+    restricted_paths: &'a HashMap<String, String>,
+    violations: Vec<Diagnostic>,
+}
+
+impl<'a> Visitor for RestrictedImportsVisitor<'a> {
+    fn visit_local_assignment(&mut self, node: &ast::LocalAssignment) {
+        if self.restricted_paths.is_empty() {
+            return;
+        }
+
+        // Check each assignment in the local statement
+        for expression in node.expressions() {
+            if let Some(path) = name_path(expression) {
+                let full_path = path.join(".");
+
+                // Check if this path is restricted
+                if let Some(message) = self.restricted_paths.get(&full_path) {
+                    let range = expression.range().unwrap();
+
+                    self.violations.push(Diagnostic::new(
+                        "restricted_imports",
+                        format!("Import path `{}` is restricted: {}", full_path, message),
```

Comment:
```suggestion
                        format!("import path `{full_path}` is restricted"),
```

For the message, add that as a note. Check out Diagnostic::new_complete